### PR TITLE
Disallow --serve in test mode

### LINKED ISSUE #40: Using `--serve` in test mode does not serve the test runner
## Enhancement

At the moment using the `--serve` option when building in test mode is pretty useless as it serves the `output/test` directory meaning that it does not enable running the tests in the browser.

We should consider serving from the project root or disabling `--serve` when building the tests.

### PR DESCRIPTION
**Pull request by mwistrand**
Resolves #40 (and part of #38). Throws an error if the user attempts to serve in test mode.

### CODE REVIEW

File: test-app/fixtures/windows/dist/index.html [hunk 1]
```diff
@@ -28,13 +25,16 @@
 		var element = document.getElementById('app');
 		var target;
 		paths.some(function (path, i) {
-			target = html[i];
-			return path && ((typeof path === 'string' && path === window.location.hash) || (typeof path === 'object' && path.match && new RegExp(path.match.join('|')).test(window.location.hash)));
+			var match = (typeof path === 'string' && path === window.location.hash) || path && (typeof path === 'object' && path.match && new RegExp(path.match.join('|')).test(window.location.hash));
```
Comment (agubler) [inline test-app/fixtures/windows/dist/index.html]: surprised i didn't have to regenerate these when I made the change!

File: tests/unit/main.ts [hunk 2]
```diff
@@ -264,6 +264,18 @@ describe('command', () => {
 			});
 		});
 
+		it('should disallow serve in test mode', () => {
+			const main = mockModule.getModuleUnderTest().default;
+			main
```
Comment (agubler) [inline tests/unit/main.ts]: do you not need to return the promise?

Comment (mwistrand) [reply tests/unit/main.ts]: Nice catch. Fixed.

**Comment by agubler** [comment]
@mwistrand seems the CLA bot isn't overly happy!

**Comment by mwistrand** [comment]
That's what I get for changing my GitHub email address. 😫

**Pull request merged**